esphome/components/esp32/__init__.py

```diff
@@ -80,6 +80,23 @@ def get_esp32_variant(core_obj=None):
     return (core_obj or CORE).data[KEY_ESP32][KEY_VARIANT]
 
 
+def get_download_types(storage_json=None):
```

Comment:
This should be required (or it will crash the below code)
```suggestion
def get_download_types(storage_json):
```